src/main/java/com/synopsys/integration/alert/channel/event/ChannelEventFactory.java

```diff
@@ -157,30 +159,36 @@ public EmailChannelEvent createEmailChannelTestEvent(final Config restModel) {
         if (BooleanUtils.toBoolean(emailDistributionConfig.getFilterByProject())) {
             emailDistributionConfig.getConfiguredProjects()
                 .stream()
-                .forEach(project -> emailAddresses.addAll(getEmailAddressesForProject(project)));
+                .forEach(project -> emailAddresses.addAll(getEmailAddressesForProject(project, emailDistributionConfig.getProjectOwnerOnly())));
 
         } else if (emailDistributionConfig.getProviderName().equals(BlackDuckProvider.COMPONENT_NAME)) {
-            blackDuckUserRepositoryAccessor.readEntities()
+            blackDuckProjectRepositoryAccessor.readEntities()
                 .stream()
-                .map(databaseEntity -> (BlackDuckUserEntity) databaseEntity)
-                .forEach(blackDuckUserEntity -> emailAddresses.add(blackDuckUserEntity.getEmailAddress()));
+                .map(databaseEntity -> (BlackDuckProjectEntity) databaseEntity)
+                .forEach(blackDuckProjectEntity -> emailAddresses.addAll(getEmailAddressesForProject(blackDuckProjectEntity.getName(), emailDistributionConfig.getProjectOwnerOnly())));
         }
         return new EmailChannelEvent(RestConstants.formatDate(new Date()), emailDistributionConfig.getProviderName(), emailDistributionConfig.getFormatType(), messageContent,
             null, emailAddresses, emailDistributionConfig.getEmailSubjectLine());
     }
 
-    private Set<String> getEmailAddressesForProject(final String projectName) {
+    private Set<String> getEmailAddressesForProject(final String projectName, boolean projectOwnerOnly) {
         if (StringUtils.isBlank(projectName)) {
             return Collections.emptySet();
         }
         final BlackDuckProjectEntity blackDuckProjectEntity = blackDuckProjectRepositoryAccessor.findByName(projectName);
-        final List<UserProjectRelation> userProjectRelations = userProjectRelationRepositoryAccessor.findByBlackDuckProjectId(blackDuckProjectEntity.getId());
-        final Set<String> emailAddresses = userProjectRelations
-                                               .stream()
-                                               .map(userProjectRelation -> blackDuckUserRepositoryAccessor.readEntity(userProjectRelation.getBlackDuckUserId()))
-                                               .filter(userEntity -> userEntity.isPresent())
-                                               .map(userEntity -> ((BlackDuckUserEntity) userEntity.get()).getEmailAddress())
-                                               .collect(Collectors.toSet());
+        final Set<String> emailAddresses;
+        if (projectOwnerOnly && StringUtils.isNotBlank(blackDuckProjectEntity.getProjectOwnerEmail())) {
+            emailAddresses = new HashSet<>();
+            emailAddresses.add(blackDuckProjectEntity.getProjectOwnerEmail());
+        } else {
```

Comment:
If they configured to send alerts to the Project owner only and there is none, should we not send anything? Or should we send it to the Users assigned to the project?
Right now the code is set to send it to the Users assigned to the project if there is no project owner

Reply:
Right, I meant somewhere in the configuration, so the test event would work.